src/main/java/pl/sda/library/infrastructure/jpa/BookstoreRepositoryAdapter.java

```diff
@@ -0,0 +1,54 @@
+package pl.sda.library.infrastructure.jpa;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import pl.sda.library.domain.model.Bookstore;
+import pl.sda.library.domain.port.BookstoreRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@AllArgsConstructor
+public class BookstoreRepositoryAdapter implements BookstoreRepository {
+
+    private BookstoreDtoRepository bookstoreDtoRepository;
+
+    @Override
+    public Long add(Bookstore bookstore) {
+        return bookstoreDtoRepository.save(BookstoreDto.fromDomain(bookstore)).getIdBookstore();
+    }
+
+    @Override
+    public Optional<Bookstore> edit(Bookstore bookstore) {
+        return Optional.of(bookstoreDtoRepository.save(BookstoreDto.fromDomain(bookstore)).toDomain());
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try {
+            bookstoreDtoRepository.deleteById(id);
```

Comment:
jestes pewien ze to rzuca wyjatek?

Reply:
wedlug dokumentacji nie poleci tutaj wyjatek

Reply:
przy pierwszym usunieciu zwraca 1
przy 2
{
    "timestamp": "2020-11-15T12:21:26.900+00:00",
    "status": 500,
    "error": "Internal Server Error",
    "message": "",
    "path": "/api/bookstores/2"
}

Reply:
przy takiej zmianie
 public void delete(Long id) {
        bookstoreDtoRepository.deleteById(id);
    }